Makefile

```diff
@@ -105,6 +105,10 @@ endif
 #
 #####
 
+HAS_CRANELIFT := 0
+HAS_LLVM := 0
+HAS_SINGLEPASS := 0
```

Comment:
I have a preference for naming these `ENABLE_` and then using `?=` so that the user can override and specify `ENABLE_CRANELIFT=1 make` on a new or different OS or CPU which we don't yet support with autodetection.

Reply:
OK, that's a new feature then. The `HAS_*` “constants” are set if we detect the specific compiler. If we want the user to set them, then we should also update the `compilers` variable to include the compiler identifier for the `build` and `test` rules.

I'm updating the code accordingly. It's outside the goal of this PR but I think it's an interesting feature.

Reply:
Done.

We can use `ENABLE_CRANELIFT=1 ENABLE_LLVM=0 make build-capi` to force to enable Cranelift and to force to disable LLVM when running `build-capi`. I've tested all the combinations, and it works.